packages/back-end/src/controllers/datasources.ts

```diff
@@ -513,25 +513,15 @@ export async function putDataSource(
 
   try {
     const updates: Partial<DataSourceInterface> = {
+      name,
+      description,
+      settings,
+      projects,
       dateUpdated: new Date(),
     };
 
-    if (name) {
-      updates.name = name;
-    }
-    if (description) {
```

Comment:
This is where the initial issue was happening - in the event a description was removed, this check was preventing us from actually updating the data source's description.